### PR TITLE
Compat users upgrade from Lawnchair 13

### LINKED ISSUE #3881: [BUG] SQLiteException: near "DROP": syntax error (code 1 SQLITE_ERROR)
### Describe the bug

Upgrade from LC 13:

```
android.database.sqlite.SQLiteException: near "DROP": syntax error (code 1 SQLITE_ERROR): , while compiling: ALTER TABLE favorites DROP COLUMN iconPackage;
  at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
  at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1045)
  at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:652)
  at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
  at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
  at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:33)
  at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1919)
  at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1841)
  at com.android.launcher3.provider.LauncherDbUtils.migrateLegacyShortcuts(LauncherDbUtils.java:175)
  at com.android.launcher3.model.DatabaseHelper.onUpgrade(DatabaseHelper.java:280)
  at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:416)
  at android.database.sqlite.SQLiteOpenHelper.getReadableDatabase(SQLiteOpenHelper.java:340)
  at com.android.launcher3.model.ModelDbController.createDatabaseHelper(ModelDbController.java:119)
  at com.android.launcher3.model.ModelDbController.createDbIfNotExists(ModelDbController.java:100)
  at com.android.launcher3.model.ModelDbController.migrateGridIfNeeded(ModelDbController.java:284)
  at com.android.launcher3.model.ModelDbController.tryMigrateDB(ModelDbController.java:265)
  at com.android.launcher3.model.LoaderTask.loadWorkspaceImpl(LoaderTask.java:375)
  at com.android.launcher3.model.LoaderTask.loadWorkspace(LoaderTask.java:349)
  at com.android.launcher3.model.LoaderTask.run(LoaderTask.java:212)
  at android.os.Handler.handleCallback(Handler.java:938)
  at android.os.Handler.dispatchMessage(Handler.java:99)
  at android.os.Looper.loop(Looper.java:223)
  at android.os.HandlerThread.run(HandlerThread.java:67)
```

### Steps to reproduce

1. Install LC 13 and modify homescreen.
2. Upgrade to LC 14.

### Expected behavior

No Crash.

### Screenshots

_No response_

### Device information

- Device: Emulator
- OS: Stock Android 11.
- App version: 14

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Goooler**
## Description

- Closes #3881.

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:white_check_mark: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by bnhlll** [comment]
Will there be a debug for a13?

**Comment by Goooler** [comment]
I'm fixing this issue for the releases in my fork.

**Comment by bnhlll** [comment]
Thank you very much)

File: src/com/android/launcher3/provider/LauncherDbUtils.java [hunk 1]
```diff
@@ -171,9 +173,13 @@ public static void migrateLegacyShortcuts(Context context, SQLiteDatabase db) {
...
 
+        // Compat users upgrade from Lawnchair 13.
+        removeColumn(db, Favorites.TABLE_NAME, "iconPackage");
+        removeColumn(db, Favorites.TABLE_NAME, "iconResource");
+
         // Drop the unused columns
-        db.execSQL("ALTER TABLE " + Favorites.TABLE_NAME + " DROP COLUMN iconPackage;");
-        db.execSQL("ALTER TABLE " + Favorites.TABLE_NAME + " DROP COLUMN iconResource;");
+//        db.execSQL("ALTER TABLE " + Favorites.TABLE_NAME + " DROP COLUMN iconPackage;");
+//        db.execSQL("ALTER TABLE " + Favorites.TABLE_NAME + " DROP COLUMN iconResource;");
```
Comment (MrSluffy) [inline src/com/android/launcher3/provider/LauncherDbUtils.java]: Please don't commented this, just surround this with try catch block and `Throwable ` with  
`removeColumn(db, Favorites.TABLE_NAME, "iconPackage");`
`removeColumn(db, Favorites.TABLE_NAME, "iconResource");`

**Review by MrSluffy — APPROVED** [review]

**Pull request merged**